tests/test_yaml_inputs.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+import copy
+import os
+import sys
+import warnings
+
+import omegaconf
+import pytest
+from omegaconf import DictConfig
+from omegaconf import OmegaConf as om
+
+# Add repo root to path so we can import scripts and test it
+repo_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.append(repo_dir)
+
+from scripts.train.train import main  # noqa: E402
+
+
+class TestTrainingYAMLInputs:
+    """Validate and tests error handling for the input YAML file."""
+
+    @pytest.fixture
+    def cfg(self) -> DictConfig:
+        """Create YAML cfg fixture for testing purposes."""
+        conf_path: str = os.path.join(
+            repo_dir, 'scripts/train/yamls/pretrain/testing.yaml')
+        with open(conf_path, 'r', encoding='utf-8') as config:
+            test_cfg = om.load(config)
+        assert isinstance(test_cfg, DictConfig)
+        return test_cfg
+
+    def test_mispelled_mandatory_params_fail(self, cfg: DictConfig) -> None:
+        """Check that mandatory mispelled inputs fail to train."""
+        cfg.trai_loader = cfg.pop('train_loader')
+        with pytest.raises(omegaconf.errors.ConfigAttributeError):
+            main(cfg)
+
+    def test_missing_mandatory_parameters_fail(self, cfg: DictConfig) -> None:
+        """Check that missing mandatory parameters fail to train."""
+        mandatory_params = [
+            'train_loader',
+            'model',
+            'tokenizer',
+            'optimizer',
+            'scheduler',
+            'max_duration',
+            'eval_interval',
+            'precision',
+            'max_seq_len',
+        ]
+        for param in mandatory_params:
+            orig_param = cfg.pop(param)
+            with pytest.raises(
+                (omegaconf.errors.ConfigAttributeError, RuntimeError)):
```

Comment:
Why is this either a config attribute error or a runtime error?

Reply:
It's a config error if we use the variable in `validate_config`. It's a runtime error if we `pop_config` when the key is missing and must_exist=True

https://github.com/mosaicml/llm-foundry/blob/90132f035d00de9064d3246553ac770fbce20ce1/llmfoundry/utils/config_utils.py#L29-L31